lib: bm_spi_mngr: sample: port SPI transaction manager to NCS bare metal, add sample and board configs (#795)
nrfconnect/sdk-nrf-bm: samples/peripherals/spi_mngr/README.rst

.. _spi_mngr_sample:

SPI Manager
###########

.. contents::
   :local:
   :depth: 2

> The SPI manager sample demonstrates how to use the :ref:`lib_bm_spi_mngr` library with |BMlong| to perform non-blocking read, page program, and sector erase operations on the external flash memory of the development kit.

Reviewer: ```suggestion
The SPI manager sample demonstrates how to use the :ref:`lib_bm_spi_mngr` library with |BMlong| to perform read, page program, and sector erase operations on the external flash memory of the development kit.
```

Author: fixed

Requirements
************

This sample is designed to run on a development kit with **on-board SPI external flash**, which is configured through the ``BOARD_EXTERNAL_MEMORY_*`` macros in :file:`board-config.h`.
Currently, only the `nRF54L15 DK`_ and `nRF54LM20 DK`_ include on-board external flash and are therefore the supported targets for this sample.

> The following board targets match :file:`sample.yaml`:

Reviewer: ```suggestion
The sample supports the following development kits with **on-board SPI external flash**:
```
`BOARD_EXTERNAL_MEMORY_*` and `board-config.h` is mentioned later, so I don't think we have to go over that here.

Author: ok, fixed

   However, the :ref:`lib_bm_spi_mngr` library itself is fully supported on the nRF54lv10 and nRF54ls05 SoCs.
   You can still use the SPI manager feature on these boards by connecting your own external SPI slave device (for example, an external NOR flash) to the SPI pins of the SoC and adapting the sample configuration accordingly (pin assignments in :file:`board-config.h`, command set, and timing parameters to match your device).

.. important::
   Before flashing the sample, you must enable **External memory** using the `Board Configurator`_ app in `nRF Connect for Desktop`_, and then write the configuration to the board.
   Without this step, the on-board flash is not powered or routed to the SoC, and the sample will not work.

Overview
********

The sample uses the SPI manager library to schedule read, program, and erase operations on the external flash.
All operations target the same flash address (``FLASH_ADDR`` in :file:`main.c`) and share a single SPI manager instance.

The flash is the on-board MX25-class NOR device on the development kit.

> Its SPI and strap pins are defined by the ``BOARD_EXTERNAL_MEMORY_*`` macros in :file:`board-config.h`.

Reviewer: What is a strap pin?

Author: A strap pin is a pin held at a fixed logic level to set a hardware mode (like write-protect or reset state), rather than one used for data transfer. They exist because many ICs need some way to configure fixed behavior at power on, and using a dedicated pin for this is simpler and cheaper than needing a software configuration interface.

Rewrote the sentence to explain this concept better.

.. _spi_mngr_sample:

SPI manager external memory

> ###########################

Reviewer: ```suggestion
SPI Manager
###########
```

Author: Fixed

Requirements
************

This sample is designed to run on a development kit with **on-board SPI external flash**, which is configured through the ``BOARD_EXTERNAL_MEMORY_*`` macros in :file:`board-config.h`.
Currently, only the `nRF54L15 DK`_ and `nRF54LM20 DK`_ include on-board external flash and are therefore the supported targets for this sample.

The following board targets match :file:`sample.yaml`:

.. tabs::

   .. group-tab:: Simple board variants

      The following board variants do **not** have DFU capabilities:

      .. include:: /includes/supported_boards_with_external_flash_memory_non-mcuboot_variants_s115.txt

      .. include:: /includes/supported_boards_with_external_flash_memory_non-mcuboot_variants_s145.txt

   .. group-tab:: MCUboot board variants

      The following board variants have DFU capabilities:

      .. include:: /includes/supported_boards_with_external_flash_memory_mcuboot_variants_s115.txt

      .. include:: /includes/supported_boards_with_external_flash_memory_mcuboot_variants_s145.txt

.. note::
   The `nRF54LV10 DK`_ and `nRF54LS05 DK`_ do **not** have on-board external flash memory and are therefore out of scope for this sample.

>    However, the :ref:`lib_bm_spi_mngr` library itself is fully supported on these devkits.

Reviewer 2: ```suggestion
   However, the :ref:`lib_bm_spi_mngr` library itself is fully supported on the nRF54lv10 and nRF54ls05 SoCs.
```

Author: fixed

   You can still use the SPI manager feature on these boards by connecting your own external SPI slave device (for example, an external NOR flash) to the SPI pins of the SoC and adapting the sample configuration accordingly (pin assignments in :file:`board-config.h`, command set, and timing parameters to match your device).

.. important::

>    Before flashing the sample, you must enable **external storage** using the `Board Configurator`_ app in `nRF Connect for Desktop`_, and then write the configuration to the board.

Reviewer 3: In Board Configurator v1.2.0 the *external storage* is referred to as *External memory*.
```suggestion
   Before flashing the sample, you must enable **External memory** using the `Board Configurator`_ app in `nRF Connect for Desktop`_, and then write the configuration to the board.
```

Author: fixed

   Without this step, the on-board flash is not powered or routed to the SoC, and the sample will not work.

Overview
********

> The sample uses the SPI manager library to schedule read, program, and erase operations on the external flash without blocking the CPU.

Reviewer: ```suggestion
The sample uses the SPI manager library to schedule read, program, and erase operations on the external flash.
```
Not sure that this information is valuable or that it makes sense to mention this. To me, if something is scheduled, I would really be surprised if that blocked the CPU. Any thoughts?

Author: fixed
